packages/arb-avm-cpp/data_storage/src/aggregator.cpp

```diff
@@ -186,15 +176,63 @@ uint64_t AggregatorStore::blockCount() const {
     return blockCountImpl(*tx);
 }
 
+void AggregatorStore::saveMessageBatch(const uint256_t& batchNum,
+                                       const uint64_t& logIndex) {
+    auto tx = data_storage->beginTransaction();
+    auto full_key = messageBatchKey(batchNum);
+    auto index_value = uint64Value(logIndex);
+
+    auto status = tx->Put(vecToSlice(full_key), vecToSlice(index_value));
+    if (!status.ok()) {
+        throw std::runtime_error("failed to save");
+    }
+
+    commitTx(*tx);
+}
+
+uint64_t AggregatorStore::getMessageBatch(const uint256_t& batchNum) {
+    auto tx = data_storage->beginTransaction();
+    auto full_key = messageBatchKey(batchNum);
+
+    std::string value;
+    auto status =
+        tx->GetForUpdate(rocksdb::ReadOptions{}, vecToSlice(full_key), &value);
```

Comment:
I think `Get` here is fine because we're just gonna return this value and close the tx

Reply:
done